test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs

```diff
@@ -42,6 +42,47 @@ public SamplesEndToEndTests(TestFixture fixture)
 
         public object AuthorizationLevelAttribute { get; private set; }
 
+        [Fact]
+        public async Task SetHostState_Offline_Succeeds()
+        {
+            // verify host is up and running
+            var response = await GetHostStatusAsync();
+            var hostStatus = response.Content.ReadAsAsync<HostStatus>();
+
+            // verify functions can be invoked
+            await InvokeAndValidateHttpTrigger("HttpTrigger");
+
+            // take host offline
+            await SetHostStateAsync("offline");
+
+            // when testing taking the host offline doesn't seem to stop all
+            // application services, so we issue a restart
+            await RestartHostAsync();
```

Comment:
TODO: I can see when the test runs that `IScriptJobHostEnvironment.Shutdown` is called, however that doesn't cause the WebJobsScriptHostService to start back up. However when running for real (e.g. IIS Express) Shutdown causes the app to restart as expected.